src/rpc/daemon_messages.cpp

```diff
@@ -34,99 +34,47 @@ namespace cryptonote
 
 namespace rpc
 {
+void GetHeight::Request::doToJson(rapidjson::Writer<rapidjson::StringBuffer>& dest) const
+{}
 
-const char* const GetHeight::name = "get_height";
-const char* const GetBlocksFast::name = "get_blocks_fast";
-const char* const GetHashesFast::name = "get_hashes_fast";
-const char* const GetTransactions::name = "get_transactions";
-const char* const KeyImagesSpent::name = "key_images_spent";
-const char* const GetTxGlobalOutputIndices::name = "get_tx_global_output_indices";
-const char* const SendRawTx::name = "send_raw_tx";
-const char* const SendRawTxHex::name = "send_raw_tx_hex";
-const char* const StartMining::name = "start_mining";
-const char* const StopMining::name = "stop_mining";
-const char* const MiningStatus::name = "mining_status";
-const char* const GetInfo::name = "get_info";
-const char* const SaveBC::name = "save_bc";
-const char* const GetBlockHash::name = "get_block_hash";
-const char* const GetLastBlockHeader::name = "get_last_block_header";
-const char* const GetBlockHeaderByHash::name = "get_block_header_by_hash";
-const char* const GetBlockHeaderByHeight::name = "get_block_header_by_height";
-const char* const GetBlockHeadersByHeight::name = "get_block_headers_by_height";
-const char* const GetPeerList::name = "get_peer_list";
-const char* const SetLogLevel::name = "set_log_level";
-const char* const GetTransactionPool::name = "get_transaction_pool";
-const char* const HardForkInfo::name = "hard_fork_info";
-const char* const GetOutputHistogram::name = "get_output_histogram";
-const char* const GetOutputKeys::name = "get_output_keys";
-const char* const GetRPCVersion::name = "get_rpc_version";
-const char* const GetFeeEstimate::name = "get_dynamic_fee_estimate";
-const char* const GetOutputDistribution::name = "get_output_distribution";
-
-
-
-
-rapidjson::Value GetHeight::Request::toJson(rapidjson::Document& doc) const
+void GetHeight::Request::fromJson(const rapidjson::Value& val)
 {
-  return Message::toJson(doc);
 }
 
-void GetHeight::Request::fromJson(rapidjson::Value& val)
+void GetHeight::Response::doToJson(rapidjson::Writer<rapidjson::StringBuffer>& dest) const
 {
+  INSERT_INTO_JSON_OBJECT(dest, height, height);
```

Comment:
Since it seems that in each and every instance the struct member and the field in the json have the same name, would it make sense to have a macro for this specific case? It'd be nice if we don't have to keep repeating field names everywhere.

Reply:
Yes, this was/is rather annoying. I have a macro that allows header/cpp split and brings it closer to epee style already, but I haven't changed these. This change is definitely worth a separate PR because it will incur lots of changes.